### ISSUE TITLE
Config version: desktop.nix, wm.nix

### ISSUE THREAD

**Issue by s0me1newithhand7s** [s0me1newithhand7s/myNixConf#4]
# Purpose
issue for my work on testing desktop version with different WM.
# Syntaxis
- no amd.nix/nvidia.nix bc it will be on VM.
- edits in config with screen and code.
- source to manual like nixos wiki.
# TODO
1. Hyprland - ✅ 
2. sway - ✅ 
3. swayfx  - ✅
4. labwc - ✅
5. any xorg/xserver wm. - ✅
6. end tests, leave issue opened. - ✅

**Comment by s0me1newithhand7s** [comment]
# Issue 1 in tree #4 
### Config:
- Oracle VM VBox Manager
- added `programs.swayfx.enable = true;` in `desktop.nix`
- https://nixos.wiki/wiki/Swayfx and https://nixos.wiki/wiki/Sway
### Result / Error: 
`programs.swayfx.enable = true;` does not exist. only `environment.systemPackages = [ pkgs.swayfx ]` works. 
no build errors; no syntaxis error;
### Screenshots:
--- 
(not provided as it isn't config/syntaxis error)
# TL;DR
swayfx pkg have no toggle for enabling with system, unlike sway. 
further test with sway and swayfx will be merged in one issue.

**Comment by s0me1newithhand7s** [comment]
# Issue 2 in tree #4 
### Config
- Oracle VM VBox Manager
- toggled `programs.sway.enabled` as `true`; minor changes in `desktop.nix`
- https://nixos.wiki and https://search.nixos.org/packages?channel=unstable as main sources. won't change even for #3 tree.
### Result / Error
no build errors, no syntaxis errors, no toggle errors;
works fine!
### Screenshots
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/784996bc-3ce1-4f77-a911-dc109a5d62aa)
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/494e18d7-a295-4ab3-bbdf-14cf5ab96ff8)

# TL;DR
works fine.

**Comment by s0me1newithhand7s** [comment]
# Issue 3 in tree #4 
### Config
- Oracle VM VBox Manager
- toggled `programs.sway.enabled` as `false`; 
- https://nixos.wiki/ and https://search.nixos.org/packages?channel=unstable
### Result / Error
there is no `programs.swayfx` toggles in NixOS, to provide `swayfx` with system you have to add it in `home packages` or `environment packages`. also manual configuration is needed.
 ### Screenshots
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/a010b5bd-4e09-46ac-add7-0ef75b820a4f)

# TL;DR
works, but not as good as sway. sadly :( 
(yes it **IS** swayfx, i've double checked)

**Comment by s0me1newithhand7s** [comment]
# Issue 5 in tree #4 
### Config
- Oracle VM VBox Manager
- toggled `programs.labwc.enabled` as `true`; 
- https://nixos.wiki/ and https://search.nixos.org/packages?channel=unstable
### Result / Error
i was confused at first start xDD
i won't use LabWC as my main but oh god, it's good. not dissapointed at all.
### Screenshots
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/eb23181a-dc4f-4de8-85f3-d05f3a1ad3d6)
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/471a141b-2926-45c4-b32f-2b4b659120f6)
# TL;DR
work and works fine. recommend before using it read [the manual](https://labwc.github.io/getting-started.html).

**Comment by s0me1newithhand7s** [comment]
# Issue 6 in tree #4
### Config
- Oracle VM VBox Manager
- toggled `programs.hyprland.enabled` as `true`; slightly edited hypland's config bc i use `foot`, not `kitty`. (i won't edit it more, nuh uh)
- https://nixos.wiki/ and https://search.nixos.org/packages?channel=unstable
### Result / Error
no build errors, no syntaxis errors, no toggle errors;
### Screenshots
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/a002aa04-bf23-409c-aa78-53f0585d1c6c)
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/e4e38307-4570-464c-af9e-6bd470492acc)

# TL; DR
bruh i love how hyprland is just being *hyprland*. no, it's works fine but.. this scale bug is hilarious. XDD

**Comment by s0me1newithhand7s** [comment]
# Issue 6 and last in tree #4
### Config
- Oracle VM VBox Manager
- toggled `services.xserver.windowManager.ratpoison.enable` as `true`; minor edits in `desktop.nix`
- https://nixos.wiki/ and https://search.nixos.org/packages?channel=unstable
### Result / Error
no build errors, no syntaxis errors, no toggle errors;
### Screenshots
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/cbdebf0c-409a-4047-8222-5778299d01a4)
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/2fe3b929-21e2-48bc-8af2-1f537b77f012)
![image](https://github.com/s0me1newithhand7s/myNixConf/assets/117505144/57019706-af03-465a-ae90-786bb4d342b3)

# TL;DR
what the hell is `ratpoison`. it's WAY harder than `awesome`, `xmonad`, `hurbstluft` and `dwm` lol.

**Comment by s0me1newithhand7s** [comment]
## TODO 
achieved
## Issue
closed
